client/src/api/common/data.ts

```diff
@@ -0,0 +1,69 @@
+import { useState } from "react";
+import { Array, Null, Record, Runtype, Static, String, Union, Unknown } from "runtypes";
+import { SensorsT } from "types/data";
```

Comment:
I'm not an expert on this but when dealing with typescript types are we able to use `[filename].d.ts` to automatically import the types? [This article goes through global type declarations](https://medium.com/jspoint/typescript-type-declaration-files-4b29077c43). Probably not for this PR, but just something I've noticed.

Reply:
I've had a look at the article and it does seem like we can do that. I'm not sure how importing types between files would work out (I'm thinking like C header files) because we do have some duplicated types that mainly differ between v2, v3, common. I'd say it's something interesting to look at in the future!